pkg/helpers/struct.go

```diff
@@ -140,18 +139,29 @@ func AsMap(x *messages.Struct) map[string]interface{} {
 // converted as strings to remain compatible with MarshalJSON.
 func AsInterface(x *messages.Value) interface{} {
 	switch v := x.GetKind().(type) {
-	case *messages.Value_NumberValue:
+	case *messages.Value_Float64Value:
 		if v != nil {
-			switch {
-			case math.IsNaN(v.NumberValue):
-				return "NaN"
-			case math.IsInf(v.NumberValue, +1):
-				return "Infinity"
-			case math.IsInf(v.NumberValue, -1):
-				return "-Infinity"
-			default:
-				return v.NumberValue
-			}
+			return v.Float64Value
```

Comment:
Should the NaN and Infinity cases be added here?

Reply:
So, my thinking was that it doesn't make too much sense for our use case, as most of the time in beats, we wouldn't want a float type quietly converted to a string, as the original code does.

Reply:
In my opinion the strategy we use here should be to pick the format that Elasticsearch accepts, or to let Elasticsearch be the ultimate decision maker on if the data format is acceptable.

ES doesn't look like it accepts NaN and Inf/-Inf values, so the best approach might be to just pass these numbers through as is and let the indexing error tell the user the data is malformed. https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html.

Converting to string here is likely to cause less obvious problems in production if the data destined for a field mapped as double and we tried to write a string value. I agree with not doing that anymore.

Another option is to detect these values in the shipper and not even attempt to index them with an obvious error message. I'm not sure if there is value in doing that here though, we can wait and see how common this problem is in reality and special case it later.

Reply:
Thank you both.  Makes sense to pass through the value and let the downstream system (try to) handle it.